imports/plugins/included/product-variant/components/productGrid.js

```diff
@@ -205,22 +203,89 @@ class ProductGrid extends Component {
               )}
             </ConfirmDialog>
 
+            <ConfirmDialog
+                title={i18next.t("admin.productTable.bulkActions.archiveTitle", { count })}
+                message={i18next.t("admin.productTable.bulkActions.archiveMessage")}
+                onConfirm={this.handleBulkActionArchive}
+              >
+                {({ openDialog }) => (
+                  <MenuItem onClick={openDialog}>{i18next.t("admin.productTable.bulkActions.archive")}</MenuItem>
+                )}
+              </ConfirmDialog>
+          </Menu>
+        </Toolbar>
+      );
+    } else {
+      return (
+        <Toolbar>
+          <Button
+            aria-owns={bulkActionMenuAnchorEl ? "bulk-actions-menu" : undefined}
+            aria-haspopup="true"
+            onClick={this.handleShowBulkActions}
+            variant="outlined"
+          >
+            {i18next.t("admin.productTable.bulkActions.actions")}
+            <ChevronDownIcon />
+          </Button>
+          <Menu
+            id="bulk-actions-menu"
+            anchorEl={bulkActionMenuAnchorEl}
+            open={Boolean(bulkActionMenuAnchorEl)}
+            onClose={this.handleCloseBulkActions}
+          >
+            <ConfirmDialog
+              title={i18next.t("admin.productTable.bulkActions.publishTitle", { count })}
+              message={i18next.t("admin.productTable.bulkActions.publishMessage")}
+              onConfirm={this.handleBulkActionPublish}
+            >
+              {({ openDialog }) => (
+                <MenuItem disabled onClick={openDialog}>{i18next.t("admin.productTable.bulkActions.publish")}</MenuItem>
+              )}
```

Comment:
This implementation is so ugly.

I wanted to do something like this, but I couldn't get it to work:

```
const isAnySelected = Array.isArray(selectedProductIds) && selectedProductIds.length;
```

```
              {({ openDialog, isAnySelected }) => (
                <MenuItem { isAnySelected ? "disabled" : "" } onClick={openDialog}>{i18next.t("admin.productTable.bulkActions.publish")}</MenuItem>
              )}
```

For `disabled`, an implementation like this doesn't work either: 

```
disabled="isAnySelected"
```

Even if `disabled="false"`, or `disabled="null"`, the MenuItem will still render as `disabled.`

Any other better ideas to make this a bit cleaner??

Reply:
PS. Trying the code out in this sandbox is cool and fast: https://catalyst.reactioncommerce.com/#/Base%20Components/Feedback/ConfirmDialog